File: src/main/java/com/ajax_systems/akka/CsvProcessor.java
package com.ajax_systems.akka;

import com.ajax_systems.akka.console.CsvConsole;
import com.ajax_systems.akka.module.ConsoleModule;
import com.ajax_systems.akka.module.WebModule;
import com.ajax_systems.akka.web.util.Cli;
import com.google.inject.Guice;
import com.google.inject.Injector;
import com.ajax_systems.akka.web.route.CsvRoute;
import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import java.util.concurrent.ExecutionException;


public class CsvProcessor {

    private static final Config config = ConfigFactory.load();

    public static void main(String... args) throws InterruptedException, ExecutionException {

        Cli cli = new Cli(args);
        cli.actions();

        switch (cli.getRun()) {
            case "console" : {
                System.out.println("You starting as CONSOLE application");
                Injector injector = Guice.createInjector(new ConsoleModule());
                injector.getInstance(CsvConsole.class)
                        .startConsole(cli.getBooksByWriter(), cli.getSort());
                System.exit(0);
                break;
            }
            case "web" : {
                System.out.println("You starting as WEB application");
                Injector injector = Guice.createInjector(new WebModule());
                injector.getInstance(CsvRoute.class)
                        .startServer(config.getString("csvRoute.host"), config.getInt("csvRoute.port"));
                break;
            }
            default:
                System.out.println("Here you can set parameter 'console' or 'web'?");
                break;
        }
    }
}

File: src/main/scala/com/ajax_systems/akka/CsvProcessor.scala
package com.ajax_systems.akka

import com.ajax_systems.akka.module.{ConsoleModule, WebModule}
import com.ajax_systems.akka.web.route.CsvRoute
import com.google.inject.{Guice}
import com.typesafe.config.{Config, ConfigFactory}
import com.ajax_systems.akka.web.util.Cli
import com.ajax_systems.akka.console.CsvConsole


object CsvProcessor extends App {

  val config: Config = ConfigFactory.load

  val cli: Cli = new Cli(args)
  cli.actions

  cli.getRun match {
    case "console" => {
      println("You starting as CONSOLE application")
      val injector = Guice.createInjector(new ConsoleModule)
      val component = injector.getInstance(classOf[CsvConsole])
        .startConsole(cli.getBooksByWriter(), cli.getSort())
      System.exit(0)
    }
    case "web" => {
      println("You starting as WEB application")
      val injector = Guice.createInjector(new WebModule)
      val component = injector.getInstance(classOf[CsvRoute])
        .startServer(config.getString("server.host"), config.getInt("server.port"))
    }
    case _ => println("Here you can set parameter 'console' or 'web'?")
  }
}
